frame/conviction-voting/src/lib.rs

```diff
@@ -0,0 +1,625 @@
+			Ok(())
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	/// Actually enact a vote, if legit.
+	fn try_vote(
+		who: &T::AccountId,
+		poll_index: PollIndexOf<T>,
+		vote: AccountVote<BalanceOf<T>>,
+	) -> DispatchResult {
+		ensure!(vote.balance() <= T::Currency::free_balance(who), Error::<T>::InsufficientFunds);
+		T::Polls::try_access_poll(poll_index, |poll_status| {
+			let (tally, class) = poll_status.ensure_ongoing().ok_or(Error::<T>::NotOngoing)?;
+			VotingFor::<T>::try_mutate(who, &class, |voting| {
+				if let Voting::Casting(Casting { ref mut votes, delegations, .. }) = voting {
+					match votes.binary_search_by_key(&poll_index, |i| i.0) {
+						Ok(i) => {
+							// Shouldn't be possible to fail, but we handle it gracefully.
+							tally.remove(votes[i].1).ok_or(ArithmeticError::Underflow)?;
+							if let Some(approve) = votes[i].1.as_standard() {
+								tally.reduce(approve, *delegations);
+							}
+							votes[i].1 = vote;
+						},
+						Err(i) => {
+							ensure!(
+								(votes.len() as u32) < T::MaxVotes::get(),
+								Error::<T>::MaxVotesReached
+							);
+							votes.insert(i, (poll_index, vote));
+						},
+					}
+					// Shouldn't be possible to fail, but we handle it gracefully.
+					tally.add(vote).ok_or(ArithmeticError::Overflow)?;
+					if let Some(approve) = vote.as_standard() {
+						tally.increase(approve, *delegations);
+					}
+				} else {
+					return Err(Error::<T>::AlreadyDelegating.into())
+				}
+				// Extend the lock to `balance` (rather than setting it) since we don't know what
+				// other votes are in place.
+				Self::extend_lock(who, &class, vote.balance());
+				Ok(())
+			})
+		})
+	}
+
+	/// Remove the account's vote for the given poll if possible. This is possible when:
+	/// - The poll has not finished.
+	/// - The poll has finished and the voter lost their direction.
+	/// - The poll has finished and the voter's lock period is up.
+	///
+	/// This will generally be combined with a call to `unlock`.
+	fn try_remove_vote(
+		who: &T::AccountId,
+		poll_index: PollIndexOf<T>,
+		class_hint: Option<ClassOf<T>>,
+		scope: UnvoteScope,
+	) -> DispatchResult {
+		let class = class_hint
+			.or_else(|| Some(T::Polls::as_ongoing(poll_index)?.1))
+			.ok_or(Error::<T>::ClassNeeded)?;
+		VotingFor::<T>::try_mutate(who, class, |voting| {
+			if let Voting::Casting(Casting { ref mut votes, delegations, ref mut prior }) = voting {
+				let i = votes
+					.binary_search_by_key(&poll_index, |i| i.0)
+					.map_err(|_| Error::<T>::NotVoter)?;
+				let v = votes.remove(i);
+
+				T::Polls::try_access_poll(poll_index, |poll_status| match poll_status {
+					PollStatus::Ongoing(tally, _) => {
+						ensure!(matches!(scope, UnvoteScope::Any), Error::<T>::NoPermission);
+						// Shouldn't be possible to fail, but we handle it gracefully.
+						tally.remove(v.1).ok_or(ArithmeticError::Underflow)?;
+						if let Some(approve) = v.1.as_standard() {
+							tally.reduce(approve, *delegations);
+						}
```

Comment:
Seems to me remove + reduce and add + increase is a common pattern, and could be wrapped into a single call to prevent forgetting about one or the other